docs/flow/reference/components/prometheus.remote_write.md

```diff
@@ -0,0 +1,349 @@
+different labels.
+
+[remote_write-spec]: https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit
+
+## Example
+
+```river
+prometheus.remote_write "staging" {
+  // Send metrics to a locally running Mimir.
+  endpoint {
+    url = "http://mimir:9009/api/v1/push"
+
+    http_client_config {
+      basic_auth {
+        username = "example-user"
+        password = "example-password"
+      }
+    }
+  }
+}
+
+// Configure a prometheus.scrape component to send metrics to
+// prometheus.remote_write component.
+prometheus.scrape "demo" {
+  targets = [
+    // Collect metrics from Grafana Agent's default HTTP listen address.
+    {"__address__" = "127.0.0.1:12345"},
+  ]
+  forward_to = [prometheus.remote_write.staging.receiver]
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`external_labels` | `map(string)` | Labels to add to metrics sent over the network. | | no
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`prometheus.remote_write`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+endpoint | [endpoint][] | Location to send metrics to. | no
+endpoint > http_client_config | [http_client_config][] | HTTP client settings when connecting to the endpoint. | no
+endpoint > http_client_config > basic_auth | [basic_auth][] | Configure basic_auth for authenticating to the endpoint. | no
+endpoint > http_client_config > authorization | [authorization][] | Configure generic authorization to the endpoint. | no
+endpoint > http_client_config > oauth2 | [oauth2][] | Configure OAuth2 for authenticating to the endpoint. | no
+endpoint > http_client_config > oauth2 > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
+endpoint > http_client_config > tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
+endpoint > queue_config | [queue_config][] | Configuration for how metrics are batched before sending. | no
+endpoint > metadata_config | [metadata_config][] | Configuration for how metric metadata is sent. | no
+wal | [wal][] | Configuration for the component's WAL. | no
+
+The `>` symbol indicates deeper levels of nesting. For example, `endpoint >
+http_client_config` refers to an `http_client_config` block defined inside a
+`endpoint` block.
+
+[endpoint]: #endpoint-block
+[http_client_config]: #endpoint-block
+[basic_auth]: #basic_auth-block
+[authorization]: #authorization-block
+[oauth2]: #oauth2-block
+[tls_config]: #tls_config-block
+[queue_config]: #queue_config-block
+[metadata_config]: #metadata_config-block
+[wal]: #wal-block
+
+### endpoint block
+
+The `endpoint` block describes a single location to send metrics to. Multiple
+`endpoint` blocks may be provided to send metrics to multiple locations.
+
+The following arguments are supported:
+
```

Comment:
> Not sure if we can can override the default table style for these tables to disable the alternating row styles? Another option would be to remove the code font like you do in the table at the top of the Blocks section but I like the code font for the arguments... Makes it easy to scan.

Yeah, I'm generally disappointed in the styling for the tables because of exactly this. I don't think we can override it, and we might have to wait a few weeks for @jdbaldry to come back before we can see if we have any options here.

I'd personally like to keep the code font for the arguments for this specific PR, but after all the docs are done, I'd like to discuss offline if we should generally avoid code fonts for tables until we have different styling in place.